rfd/0007/readme.mdx

```diff
@@ -0,0 +1,157 @@
+- No more push-to-main-and-deploy workflow
+unless the user turns on auto-approval - which is the same as the current solution.
+
+
+## Alternative Approaches Considered
+
+### SQL Generation Workflow
+
+#### Overview
+Generate SQL migration scripts as part of the development process rather than executing changes directly at runtime.
+
+#### Process
+1. **Pre-deployment Diff**: Compare against a reference version
+  (remote Moose state, or the end state of accumulated migration scripts.)
+2. **SQL Script Generation**: Produce SQL statements similar to our current solution but output to files instead of executing
+3. **User Review**: Require user to commit SQL files containing `DROP COLUMN` statements - no surprise data loss
+
+#### Benefits
+- Explicit user consent for destructive operations
+- Version control for schema changes
+- Predictable migration behavior
+
+#### Challenges
+- **Version Coordination**: If version 1 diffs against version 0, and version 2 also diffs against version 0,
+  then version 1 is deployed, version 2's SQL statements become invalid
+- **External Changes**: Need mechanisms to cope with external database modifications
+- **Rollback Complexity**: Rollbacks require new images with new SQL statements,
+  adding build time when rapid fixes are needed
+
+### Git History Integration
+
+#### Overview
+Embed git history into Docker images to maintain change lineage and enable safe redeployments.
+
+#### Process
+1. **Change Tracking**: Include git commit history in container images
+2. **State Consistency**: Use git history to determine what changes have been applied
+3. **Safe Redeployment**: Prevent dropping changes that were applied in previous versions
+
+#### Benefits
+- Solves redeployment safety issues
+- Maintains change history
+- Enables rollback tracking
+
+#### Challenges
+- Container size increase
+- Complexity in change tracking logic
+- Integration with existing CI/CD pipelines
+
+### User Override System
+
+#### Overview
+Maintain the current solution but add user-provided transformation layer for change operations.
+
+#### Process
+1. **Change Detection**: Generate a sequence of actions as in the existing behaviour
+2. **User Transform Function**: Pass the actions through user-provided `OlapChange[] => OlapChange[]` transformation
+3. **Full Flexibility**: Users can modify, reorder, or completely replace the generated changes
+
+#### Benefits
+- Maximum flexibility and user control
+- Backward compatibility with the current system
+- Handles edge cases through user customization
+
+#### Challenges
+- Requires user to anticipate changes done by moose
+- Potential for user errors in transformation logic
+
+## Open Questions
+
+1. **Update Atomicity**: How do we handle/stop concurrent writes when a migration is in progress?
+
+2. **Performance at Scale**: How do migrations perform on large tables?
+  Changing subfields in Nested/Tuple requires an `ALTER TABLE UPDATE` mutation.
+
+3. **Plan Data structure**:
```

Comment:
Maybe a mix of those?
ie lists of SQL in JSON that way we can also use the plan for the topic changes